app/src/main/java/com/varun/gbu_timetables/TimetableActivity.java

```diff
@@ -87,21 +96,39 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
 
-        TimeTableBasic info = getCurrentBasic();
+        final TimeTableBasic info = getCurrentBasic();
         reload();
 
         final Drawable fav_yes = Utility.ThemeTools.FavouriteIcon.getFavYes(getApplicationContext());
         final Drawable fav_no = Utility.ThemeTools.FavouriteIcon.getFavNo(getApplicationContext());
 
-        final FloatingActionButton fab = findViewById(R.id.fab);
+        final Drawable myclass_yes = Utility.ThemeTools.MyClassIcon.getMyClassYes(getApplicationContext());
+        final Drawable myclass_no = Utility.ThemeTools.MyClassIcon.getMyClassNo(getApplicationContext());
+
+        fab = findViewById(R.id.fab);
+        fab1 = findViewById(R.id.fab1);
+        fab2 = findViewById(R.id.fab2);
+
+        fab.setImageDrawable(Utility.ThemeTools.FabIcon.getfabup(getApplicationContext()));
+
+        fab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(!isFABOpen){
+                    showFABMenu();
+                }else{
+                    closeFABMenu();
+                }
+            }
+        });
 
         if (existing_data.contains(info)) {
-            fab.setImageDrawable(fav_yes);

```

Comment:
Since we have multiple fabs, using fab1 and fab2 is a bit confusing, can we rename them to something like FavouritesFab and MyClassFab, Android Studio has good shortcuts to rename variables.

Reply:
Yes, It can be refactored